bob-backend/src/pearl/core.rs

```diff
@@ -166,9 +166,13 @@ impl BackendStorage for Pearl {
             .iter()
             .find(|dc| dc.can_process_operation(&op));
         if let Some(disk_controller) = dc_option {
-            disk_controller
-                .put(op, key, data)
-                .await
+            let handler = tokio::spawn(
+                disk_controller.clone().put(op, key, data.clone())
```

Comment:
Probably, you should use async block to avoid requirements of changing signature of functions in disk_contoller:
```
tokio::spawn(async move {
    disk_controller.clone().put(op, key, data.clone()).await
});